front/src/components/Chat/Channel.tsx

```diff
@@ -1,52 +1,154 @@
+				<HiOutlineXMark className="close-icon" onClick={_ => props.setTrigger(false)} /> <br />
 				<h3>Channel Name</h3>
-				<input type="text" placeholder="New channel name" onChange={(e) => {setChanName(e.target.value)}}/>
+				<input type="text" id="channel-input" onChange={e => { setChanName(e.target.value) }} />
 				<h3>Channel Mode</h3>
 				<input type="radio" name="Mode" value="Public" />Public <span></span>
 				<input type="radio" name="Mode" value="Private" />Private <span></span>
 				<input type="radio" name="Mode" value="Protected" />Protected <br /><br />
 				<button onClick={() => handleCreateNewChan()}>Create Channel</button><span></span>
-				<button onClick={() => props.setTrigger(false)}>close</button>
 			</div>
 		</div>
-	) : <div></div>;
+	) : <></>;
+}
+
+function ChannelList() {
+	const [chanList, setChanList] = useState<IChannel[]>([]);
+	const [chanId, setChanId] = useState("");
+	const navigate = useNavigate();
+
+	socket.on('createPublicChannelOk', (newChanId) => {
+		getChanId(newChanId);
+	});
+
+	const getChanId = (data: any) => {
+		setChanId(data);
+	}
+	console.log('chanId: ' + chanId);
+	useEffect(() => {
+		fetch(`${process.env.REACT_APP_BACK}channel/${chanId}`)
+			.then(response => {
+				return response.json();
+			})
+			.then(data => {
+				console.log(data);
+				setChanList([...chanList, data]);
+			});
+	}, [chanId]);
+	return (
+		<div>
+			{chanList.map(chan => (
+				<ul key={chan.name}>
+					<div className="channel-name" onClick={_ => navigate(`chan${chan.id}`)}>{chan.name}</div>
+				</ul>
+			))}
+		</div>
+	);
 }
 
-export function ChannelList() {
-	const [buttonPopup, setButtonPopup] = React.useState(false);
+function JoinedChannelList() {
+	const [buttonPopup, setButtonPopup] = useState(false);
 
-	const chanList = [
-		<li>test chan 1</li>,
-	]
 	return (
-		<div className="title"> Channels <span> <button onClick={() => setButtonPopup(true)}>add</button></span>
+		<div className="title"> Joined Channels <span><HiPlusCircle className="add-icon" onClick={() => setButtonPopup(true)} /></span><hr />
 			<CreateChannelPopup trigger={buttonPopup} setTrigger={setButtonPopup} />
-			{chanList}
 		</div>
 	);
 }
+
+function PublicChannelList() {
+	return (
+		<div className="title" id="upper-side">Public Channels <hr />
+			<ul>
+				<p>test</p>
+			</ul>
+		</div>
+	);
+}
+
+function ChannelMemberList() {
+	return (
+		<div className="title"> Members <hr />
+		</div>
+	);
+}
+
+function ChannelMessages() {
+	const [newInput, setNewInput] = useState("");
+	const [messageList, setMessageList] = useState<any[]>([]);
+	const [newMessage, setNewMessage] = useState("");
+
+	const myUser = useAppSelector(state => state.user);
+
+	const handleSubmitNewMessage = (e: React.FormEvent<HTMLFormElement>) => {
+		e.preventDefault();
+		if (newInput != "")
+			socket.emit('sendMessageChannel', { chanid: 1, userid: 1, message: newInput }); //to be modified later
```

Comment:
The value is still hardcoded right now, needs to figure out a way to identify the chan and the user